### DISCUSSION TITLE
error only after deployment on vercel

### DISCUSSION THREAD

**Discussion by mohammedawez** [Jay-study-nildana/FrontEndForStudents#28] [General]
hlo jay sir

@Jay-study-nildana 

i not able to deployment my code in github,the step after git commit i m entering the commit in vs code after that courser it is not not going down to enter GITPUSH and GITSTATUS  help me with this one!!!!!!!! !!!

 i have watched your youtube video also https://www.youtube.com/watch?v=N7vtvD5DtUc

**Comment by Jay-study-nildana** [comment]
@mohammedawez 

I have made a new video, see if that helps. https://youtu.be/snF5CvFtIEk

also, please understand that cursor WILL NOT COME after 'git commit'. So, cursor will go away. You have to use the steps (given in the video), watch that step again and again. 

remember CURSOR will go away. But, you can still type. What you have to type, watch the video (both the old link that you have and also the new link). This is a step all students get stuck, but, if you are patient, you will solve it.

Let me know if were able to solve it. 

Thanks
Jay

**Reply by Jay-study-nildana** [reply]
@mohammedawez 

also, please create account on discord. 

and then, join my student discord group chat (it's like whatsapp chat) and you can ask questions with me and other students :)

Thanks

Jay.

**Reply by Jay-study-nildana** [reply]
here is the link for discord group chat server. https://sandkdesignstudio.in/fcc/

**Reply by mohammedawez** [reply]
ok sir tqqqq..... i ill try
but i have one suggesation for  in your youtube video which u hv uploded make  parts as chapter #intro  
it will be easy to watch and intresting 

suggesation from you student


@Jay-study-nildana

**Reply by Jay-study-nildana** [reply]
@mohammedawez 

I will think about it and thank you for your feedback. 

and if you have more doubts, keep asking, no problem. and learn to use discord. very useful for developers.

alright. good.

Jay.

**Reply by mohammedawez** [reply]
tq sir this step worked it maked me happy!!!


@Jay-study-nildana

**Comment by mohammedawez** [comment]
@Jay-study-nildana 

After deploy my project is not working ???? what to do sir???
here is the link    https://nasa-project-dude.vercel.app/  
this is my first deploy project i have followed each step by you https://github.com/mohammedawez/nasa-mini-project-.git (git hub link)

plz check this one sir @Jay-study-nildana

**Comment by Jay-study-nildana** [comment]
@mohammedawez 

Your code is not correct. look at this error (you have to open the developer console and see if you get any errors when checking for output)

![error1](https://github.com/Jay-study-nildana/FrontEndForStudents/assets/14297315/78cfd739-e0fe-4ff2-b6f2-eb7094509a91)

check the original NASA code, and see what this word means

```
Uncaught ReferenceError: loadNASAData is not defined
```

When building a project. 

1. First step, get the original project from Jay sir, from CodePen.
2. Repeat the project, by forking the codePen and learning and practising.
3. Then, try to build the project, in VS Code by looking at codepen.
4. make sure, VS Code, project, on computer, you are getting output.
5. Then, only when everything is working great, deploy to Vercel :) 

So, my first request is, you put your code in CodePen, and see if it works.  So, next step, I want to see your codepen project with this code. 

check my youtube channel for how to use codepen :) 

Keep me updated

Jay.

**Comment by mohammedawez** [comment]
ok sir @Jay-study-nildana  but 

i have done it again and again,that error is showing only after deploy that is displayed ,in vs code and codepen that is not displayed, whatever the coding i have learned is by your teaching onlly tqqqqq sir ,but really i dont know what to be done code is here plz see it.    https://codepen.io/mohammedawez/pen/ZEqmrwx?editors=1010  fix it
 


@Jay-study-nildana

**Reply by Jay-study-nildana** [reply]
@mohammedawez 

Excellent. That means, your code is correct, and the issue is with deployment. This is a common issue that all developers face. 

Code works in computer. But, does not work after deployment. 

Now, here are the next steps.

1. Put this code in a brand new PUBLIC github repository. 
2. Then, put that public github link here in this discussion.
3. Then, I will try to deploy it, let me see what happens when I try to deploy.

Okay. I am glad you are learning to ask questions like a developer. So, reply here once you have done the above next steps. 

Let's solve this together, yes!

Jay.

**Reply by mohammedawez** [reply]
@Jay-study-nildana 

hlo sir here is the new repositotories link https://github.com/mohammedawez/nasa-project-for-deployment  

@Jay-study-nildana

**Reply by Jay-study-nildana** [reply]
@mohammedawez 

Excellent. I will 

1. check the code in VS Code
2. Then, I will try to do a deployment.

it may take me 1 day or 3 days to do the above step. But, I will do the steps and update you.

If I don't reply in 3 days, please let me know.

Thanks.

Jay

**Comment by Jay-study-nildana** [comment]
@mohammedawez 

I have managed to fix. You can see the code here.

https://github.com/Jay-study-nildana/todeletenasaproject1

and deployment here

https://todeletenasaproject1.vercel.app/

The main issue is this line.

```
<script src="Script.js"></script> 
```

Your file name is actually "script.js". The 's' is lower case.

So, you just have to make the line like this.

```
<script src="script.js"></script> 
```

This is one of those very stupid, beginner developer mistakes that many students do. please remember that uppercase and lowercase are very important in coding, like so many other things.

The good thing, you are becoming smart by asking questions on github and taking help from tutor. So, that is how you grow as a developer.

So, I will be closing this discussion. If you have more questions, please create another issue or discussion.

Happy coding. 

Jay.
